src/instructlab/system/info.py

```diff
@@ -158,9 +167,34 @@ def get_sysinfo() -> typing.Dict[str, typing.Any]:
     return d
 
 
+def categorize_and_print(sysinfo: typing.Dict[str, typing.Any]):
+    """Automatically categorize and print system information"""
+    categories: dict[str, list[tuple[str, typing.Any]]] = {}
+
+    for key, value in sysinfo.items():
```

Comment:
Wouldn't it be better to build the desired structure from the beginning rather than create sysinfo, then reparse it into categories?